src/error.js

```diff
@@ -355,3 +367,43 @@ export function detectNonAmpJs(win) {
 export function resetAccumulatedErrorMessagesForTesting() {
   accumulatedErrorMessages = [];
 }
+
+export function detectJsEngineFromStack() {
+  const object = Object.create({
+    // DO NOT rename this property.
+    // DO NOT transform into shorthand method syntax.
+    t: function() {
+      throw new Error('message');
+    }
+  });
+  try {
```

Comment:
Firefox and IE return different results based on how the prototype is constructed.

```js
function Klass() {}
Klass.prototype.t

// Different than
proto = { t };
obj = Object.create(proto);
```

It'd have to do much more complicated testing if I don't know for certain how the chain is constructed.